plugins/linux-capture/xcomposite-input.c

```diff
@@ -302,19 +294,45 @@ xcb_window_t xcomp_find_window(xcb_connection_t *conn, Display *disp,
 	const char *thirdStr = secondMark + markSize;
 
 	// wstr only consists of the window-id
-	if (!firstMark)
+	if (!firstMark) {
+		*p_wname = NULL;
+		*p_wcls = NULL;
 		return (xcb_window_t)atol(str);
+	}
 
 	// wstr also contains window-name and window-class
 	char *wname = bzalloc(secondMark - secondStr + 1);
 	char *wcls = bzalloc(strEnd - thirdStr + 1);
 	memcpy(wname, secondStr, secondMark - secondStr);
 	memcpy(wcls, thirdStr, strEnd - thirdStr);
-	ret = (xcb_window_t)strtol(str, NULL, 10);
+	xcb_window_t ret = (xcb_window_t)strtol(str, NULL, 10);
+
+	*p_wname = wname;
+	*p_wcls = wcls;
+	return ret;
+}
+
+xcb_window_t xcomp_find_window(xcb_connection_t *conn, Display *disp,
+			       const char *str)
+{
+	xcb_window_t ret = 0;
+	char *wname = NULL;
+	char *wcls = NULL;
+	DARRAY(xcb_window_t) tlw = {0};
+
+	if (!str || strlen(str) == 0) {
+		if (tlw.num > 0)
```

Comment:
(probably fixed in later commit but,) tlw is being referenced uninitialized, the line 333 initialization is supposed to occur before this check.

Reply:
Fixed.

Reply:
Fixed!